modules/nf-core/software/rsem/preparereference/main.nf

```diff
@@ -24,15 +28,41 @@ process RSEM_PREPAREREFERENCE {
 
     script:
     def software = getSoftwareName(task.process)
-    """
-    mkdir rsem
-    rsem-prepare-reference \\
-        --gtf $gtf \\
-        --num-threads $task.cpus \\
-        $options.args \\
-        $fasta \\
-        rsem/genome
+    def args     = options.args.tokenize()
+    if (args.contains('--star')) {
+        args.removeIf { it.contains('--star') }
+        def memory = task.memory ? "--limitGenomeGenerateRAM ${task.memory.toBytes() - 100000000}" : ''
+        """
+        mkdir rsem
+        STAR \\
+            --runMode genomeGenerate \\
+            --genomeDir rsem/ \\
+            --genomeFastaFiles $fasta \\
+            --sjdbGTFfile $gtf \\
+            --runThreadN $task.cpus \\
+            $memory \\
+            $options.args2
+        
+        rsem-prepare-reference \\
+            --gtf $gtf \\
+            --num-threads $task.cpus \\
+            ${args.join(' ')} \\
+            $fasta \\
+            rsem/genome

```

Comment:
Think the name is a bit misleading for this process - also isn't this more of a subworkflow?

Reply:
Thanks alot @KevinMenden ! Unfortunately, we have to resort to this because it seems like the authors aren't going to be updating RSEM anytime soon to allow for STAR specific parameters. See https://github.com/nf-core/rnaseq/issues/511. RSEM should still work for creating other types of indices e.g. Bowtie2 etc which is why I put this logic directly here so others using the module won't have the same issues we have found already.

Reply:
Yep, figured something odd like this must be the reason. Thanks for the explanation!! :+1: